### ISSUE TITLE
Hard dependency on Sprockets prevents migration to Propshaft

### ISSUE THREAD

**Issue by sedubois** [thoughtbot/administrate#2151]
* What were you trying to do?

I am following the Propshaft [upgrade guide](https://github.com/rails/propshaft/blob/main/UPGRADING.md). After replacing Sprockets with Propshaft in my Gemfile, I expect Sprockets-related errors to disappear and Administrate views to render correctly.

* What did you end up with (logs, or, even better, example apps are great!)?

Sprockets is still invoked as it's a transitive gem dependency of Administrate, leading to `Sprockets::Railtie::ManifestNeededError: Expected to find a manifest file in app/assets/config/manifest.js`. As a workaround, an empty `app/assets/config/manifest.js` file can be kept. 

However then, I get the error `Propshaft::MissingAssetError`: `The asset 'administrate/application.css' was not found in the load path` which appears to come from `app/views/administrate/application/stylesheet.html.erb`.

It looks like propshaft and sprockets do not play along each other nicely, and Administrate should support working with Propshaft, otherwise it prevents applications from migrating.

* What versions are you running?
  - Rails 7.0.2.2
  - administrate 0.17.0 rev 3b2413acc52eee2a39eb1623dd1c867cf6bb85be

**Comment by nickcharlton** [comment]
Thanks for opening this!

Yeah, I was suspecting this might be the case. I'm not sure if we can (or maybe I mean should?) do anything about this until some other asset changes are done (I'm thinking of #2148, or an equivalent at least).

Do you have any thoughts on how we might be able to support both?

**Comment by sedubois** [comment]
At this point I haven't had thoughts on the solution on the administrate side as I'm really not familiar with the Rails asset pipeline. 

Just some background on how I got here:
- importmaps are the new default in Rails apps, but I've been having issues with them
- so I moved to jsbundling-rails to solve these issues
- but this leads to apparently sourceMaps [not working](https://github.com/rails/jsbundling-rails/issues/93) when using Sprockets
- because of this, our crash reports don't have proper stack traces
- so I experimented migrating to Propshaft, which is described as not yet fully stable but the successor to Sprockets
- this first requires using cssbundling-rails, however it seems to have [issues with vendored Sass stylesheets](https://github.com/rails/cssbundling-rails/issues/81)
- and is incompatible with Administrate, hence this issue report

So I'm blocked by these two last points from using Propshaft, meaning that we currently seem stuck with hard to interpret JS crash reports.

**Comment by sedubois** [comment]
@nickcharlton If I take the example of Stimulus (which is now part of the default Rails 7 stack), it's provided as a node package (`@hotwired/stimulus`). So a solution might be to publish an administrate node package with the relevant assets. I guess this should work for both JS and CSS. Or maybe something is possible with engines, which I know nothing about...

(NB: There is a `stimulus-rails` gem which does directly provide JS files, but they seem only usable when using Sprockets. Otherwise, all the gem does is provide a `rails stimulus:install` task to install a reference to the node package.)

**Comment by nickcharlton** [comment]
Thanks! That's helpful.

I think our eventual solution here is going to be to split it into an npm package too. I've been thinking about how to update in such a way to support all of the asset handling approaches and I suspect this is the only way.

**Comment by adam-majkowski** [comment]
@sedubois did you succeed in making `administrate` work with `propshaft`? We are facing the same problem in our application

**Comment by sedubois** [comment]
@adam-majkowski no, I didn't get back to using propshaft, in part because of this blocker.

**Comment by vpereira** [comment]
I'm just having to drop administrate because of hard requirement on sprockets.

**Comment by nickcharlton** [comment]
I'm going to close this we've just merged in #2397, which will mean that in future release we'll be bundling assets in the Administrate Gem and so we've been able to drop any dependencies on assets from those using the gem.

I'm going to start releasing some pre-release versions of the gem in the coming days, as I'm sure I'm about to break stuff for people, and so we can get feedback and make sure there's nice migration help sorted out.
